#! /bin/sh /usr/share/dpatch/dpatch-run
## 30_sh_support.dpatch by Nobuhiro Iwamatsu <iwamatsu@nigauri.org>
##
## DP: Add sh4 port.

@DPATCH@
diff -urNad poco~/Foundation/include/Poco/Platform.h poco/Foundation/include/Poco/Platform.h
--- poco~/Foundation/include/Poco/Platform.h	2009-11-24 13:31:44.000000000 +0100
+++ poco/Foundation/include/Poco/Platform.h	2009-12-08 19:40:08.000000000 +0100
@@ -135,6 +135,7 @@
 #define POCO_ARCH_ARM     0x0a
 #define POCO_ARCH_M68K    0x0b
 #define POCO_ARCH_S390    0x0c
+#define POCO_ARCH_SH      0x0d
 
 
 #if defined(__ALPHA) || defined(__alpha) || defined(__alpha__) || defined(_M_ALPHA)
@@ -183,6 +184,13 @@
 #elif defined(__s390__)
     #define POCO_ARCH POCO_ARCH_S390
     #define POCO_ARCH_BIG_ENDIAN 1
+#elif defined(__sh__)
+	#define POCO_ARCH POCO_ARCH_SH
+	#if defined(__LITTLE_ENDIAN__)
+		#define POCO_ARCH_LITTLE_ENDIAN 1
+	#else
+		#define POCO_ARCH_BIG_ENDIAN 1
+	#endif
 #endif
 
 
